Sources/AppStoreConnectCLI/Model/BetaGroup.swift

```diff
@@ -26,12 +29,25 @@ extension HTTPClient {
 
         return self.request(endpoint)
             .tryMap { response throws -> String in
+                guard !response.data.isEmpty else {
+                    throw BetaGroupError.couldntFindBetaGroup(groupNames: [name])
+                }
+
                 guard response.data.count == 1, let id = response.data.first?.id else {
-                    throw BetaGroupError.couldntFindBetaGroup
+                    throw BetaGroupError.betaGroupNotUnique(groupNames: [name])
                 }
 
                 return id
             }
             .eraseToAnyPublisher()
     }
+
+    func betaGroupIdentifiers(matching names: [String]) throws -> AnyPublisher<[String], Error> {
+        // Find group by name sequentially
+        let requests = try names.map { (name: String) throws in
+            return try self.betaGroupIdentifier(matching: name)
+        }
+
+        return Publishers.MergeMany(requests).reduce([]) { $0 + [$1] }.eraseToAnyPublisher()
```

Comment:
Hmmm how is merge working for you? Remember this won't be ordered

Reply:
In this case, this just to get the resources of a bunch groups so the order doesn't matter for now. and if I use `ConcatenateMany` it might be very slow for some big data set